orangecontrib/timeseries/widgets/owdifference.py

```diff
@@ -30,186 +37,152 @@ class Inputs:
-            callback=self.on_changed,
-            tooltip='The value corresponds to n-th order numerical '
-                    'derivative of the series. \nThe order is fixed to 1 '
-                    'if the shift period is other than 1.')
-        gui.spin(box, self, 'shift_period', 1, 100,
-                 label='Shift:',
-                 callback=self.on_changed,
-                 tooltip='Set this to other than 1 if you don\'t want to '
-                         'compute differences for subsequent values but for '
-                         'values shifted number of spaces apart. \n'
-                         'If this value is different from 1, differencing '
-                         'order is fixed to 1.')
-        gui.checkBox(box, self, 'invert_direction',
-                     label='Invert differencing direction',
-                     callback=self.on_changed,
-                     tooltip='Influences where the series is padded with nan '
-                             'values — at the beginning or at the end.')
         self.view = view = QListView(self,
                                      selectionMode=QListView.ExtendedSelection)
-        self.model = model = VariableListModel(parent=self)
-        view.setModel(model)
-        view.selectionModel().selectionChanged.connect(self.on_changed)
-        box.layout().addWidget(view)
-        gui.auto_commit(box, self, 'autocommit', '&Apply')
+        self.model = VariableListModel(parent=self)
+        view.setModel(self.model)
+        view.selectionModel().selectionChanged.connect(self._selection_changed)
+        self.controlArea.layout().addWidget(view)
+
+        box = gui.vBox(self.controlArea, "Operation")
+        gui.radioButtonsInBox(
+            box, self, 'operation',
+            [op.name for op in self.Operations],
+            callback=self._operation_changed)
+        gui.separator(box)
+
+        sp = gui.spin(
+            box, self, 'shift_period', 1, 100, label='Shift:',
+            controlWidth=60, alignment=Qt.AlignRight,
+            callback=self.commit.deferred,
+            tooltip="Sets the distance between points; 1 for consecutive points"
+        )
+        gui.rubber(sp.box)
+
+        gui.checkBox(
+            box, self, 'invert_direction', 'Invert differencing direction',
+            callback=self.commit.deferred)
+
+        gui.auto_commit(self.buttonsArea, self, 'autocommit', '&Apply')
+
+    def _operation_changed(self):
+        self.controls.shift_period.box.setEnabled(self.operation != self.Diff2)
+        self.commit.deferred()
+
+    def _selection_changed(self):
+        self.selection = [
+            self.model.data(index)
+            for index in self.view.selectionModel().selectedRows()]
+        self.commit.deferred()
 
     @Inputs.time_series
     def set_data(self, data):
-        self.closeContext()
-        self.data = data = None if data is None else \
-                           Timeseries.from_data_table(data)
```

Comment:
I think removing this is the reason it fails now on iris. Before it converted implicitly.